server/src/main/java/se/fortnox/reactivewizard/server/RwServer.java

```diff
@@ -155,6 +158,18 @@ public static void registerShutdownDependency(Runnable blockShutdownUntil) {
         RwServer.blockShutdownUntil = blockShutdownUntil;
     }
 
+    /**
+     * Registers a Runnable to be run at shutdown of RW server in order to flush and shut down the logging.
+     *
+     * @param loggingShutdownRunner the runnable
+     */
+    public static void registerLoggingShutdown(Runnable loggingShutdownRunner) {
+        if (RwServer.loggingShutdownRunner != null && loggingShutdownRunner != null) {
```

Comment:
Is your intention that running registerLoggingShutdown(null) means that you unregister a previous hook?
I would rather expect this method to do nothing or throw an error if it was passed null, so this confuses me a bit :)

Reply:
It was not exactly my intention. Actually, I copied this registration method from the dependency shutdown hook. I did not check all the possible outcomes. But yes, you're right, calling the method with null argument will in effect unregister the handler. I'd rather agree that this does not make sense. If we do not allow to reassign a handler why should it be allowed to unassign one? That does not sound consistent.

Reply:
This code has been removed. There's no handler registration done this way.